File: vue-crud-app/src/services/TransactionService.js
import http from "../http-common";

class transactionService {
  getAll() {
    return http.get("/transaction");
  }

  get(id) {
    return http.get(`/transaction/${id}`);
  }

  create(data) {
    return http.post("/transaction", data);
  }

  update(id, data) {
    return http.put(`/transaction/${id}`, data);
  }

  delete(id) {
    return http.delete(`/transaction/${id}`);
  }

  deleteAll() {
    return http.delete(`/transaction`);
  }

  findByTitle(title) {
    return http.get(`/transaction?title=${title}`);
  }
}

export default new transactionService();

File: Bank-backend/src/main/java/com/ipnet/Bank_backend/services/TransactionService.java
package com.ipnet.Bank_backend.services;

import com.ipnet.Bank_backend.entities.Transaction;

import java.util.List;

public interface TransactionService {
    Transaction create (Transaction transaction);
    Transaction update (Transaction transaction, Long id);
    List<Transaction> listeTransaction();
    void delete (Long id);
    Transaction findById (Long id);
}
